maven-core/src/main/java/org/apache/maven/DefaultMaven.java

```diff
@@ -493,22 +505,88 @@ private void validatePrerequisitesForNonMavenPluginProjects( List<MavenProject>
         }
     }
 
-    private void validateActivatedProfiles( List<MavenProject> projects, List<String> activeProfileIds )
+    /**
+     * Get all profiles that are detected in either the projects, any parent of the projects or the settings.
```

Comment:
@elharo Your help is needed, this sentence doesn't sound like proper English. It should rather read: "Get all profiles which are neither detected in the project, any parent of the projects, nor the settings." WDYT?

Reply:
I don't think you want to switch this to a negative, unless that's what the method does? I don't know. I think what's needed here is

"Get all profiles that are detected in the projects, any parent of the projects, or the settings."

unless that's exactly backwards, in which case it can be

"Get all profiles that are not detected in the projects, any parent of the projects, or the settings."

Reply:
Thanks! I took your suggestion @elharo , it definitely sounds a lot better like this.